### ISSUE TITLE
Delay in indexing data?

### ISSUE THREAD

**Issue by hagbeck** [hbz/lobid-resources#1336]
We are missing new holdings from Alma in lobid. 

Example: \
99371020204206441 created on 2002-05-09, currently no holding data in \
https://alma.lobid.org/resources/search?q=HT021120128&format=json

**Comment by acka47** [comment]
Thanks for the report, Hans-Georg. @blackwinter, can you confirm problems with Alma updates? ( @dr0i unfortunately is on sick leave this week otherwise I'd have asked him.)

**Comment by blackwinter** [comment]
@acka47: Yes, your update process has been blocked all week. Every attempt to trigger the [webhook](https://github.com/hbz/lobid-resources/blob/master/web/app/controllers/resources/Webhook.java) results in a `423 - Locked`. But @dr0i told me initially that you (`lobid-admin`?) would be notified in such cases.

**Comment by acka47** [comment]
> But @dr0i told me initially that you (lobid-admin?) would be notified in such cases.

Hmm, this doesn't seem to work. There weren't any mails like this to lobid-admin...

**Comment by blackwinter** [comment]
Well, I don't know the exact address that is configured for the webhook. This was just a guess...

**Comment by acka47** [comment]
@fsteeg just started investigating this. This is a good test for the documentation at https://github.com/hbz/lobid-resources/wiki/Maintaining-Alma.lobid-API I guess...

**Comment by fsteeg** [comment]
Seems the update process got stuck, and kept rejecting update requests because it thought an update was running. I've added myself to the list of email recipients for that problem, restarted the app, triggered the updates and we now have DE-290 items in https://alma.lobid.org/resources/search?q=HT021120128&format=json.

**Comment by hagbeck** [comment]
Thanks! The holdings are listed and therefore they should be indexed tomorrow.

**Comment by blackwinter** [comment]
@fsteeg: You probably already know this but the process got stuck again (`423 - Locked`).

**Comment by fsteeg** [comment]
Yes, thanks. For some reason the update process keeps getting stuck during indexing, with no error or hint in `web/logs/application.log` or `web/logs/ETL.log` as far as I can see. Since the fallback server seems to be working fine, I've shut down the main server, resulting in calls to `alma.lobid.org` being routed to the fallback server, so the updates should keep coming in. We will investigate further next week.

**Comment by dr0i** [comment]
Problem is the missing of some millions of docs in the full dump export of Alma. The webhook notices the missing docs and refuses further work (i.a. updates).
To gain more resilience, these things could be done in the Webhook:
- [ ] don't refuse to index updates when building of full index has failed
- [ ] escalate errors to lobid-admin (stick to errors only, reducing the noise of mailing daily 3 INFO mails )

**Comment by dr0i** [comment]
>  The webhook notices the missing docs and refuses further work (i.a. updates).

That was just an educated guess, but it seems not to be the case. It seems to be rather one of these coincidences.
It seems that, just before the unfortunate full dump, there was a `ReceiveTimeoutTransportException`:
> INFO [org.elasticsearch.client.transport.TransportClientNodesService$SimpleNodeSampler.doSample(TransportClientNodesService.java:436)] - failed to get node info for {#transport#-1}{npJLjw2nQbKfD-L5KHLzpg}{10.9.0.12}{10.9.0.12:9300}, disconnecting...
org.elasticsearch.transport.ReceiveTimeoutTransportException: [][10.9.0.12:9300][cluster:monitor/nodes/liveness] request_id [3072959] timed out after [6462ms]

This results in a hanging thread which we will be informed by as further updates/fulldumps cannot be performed.

The  `TimeoutTransportException` _could_ be the result of one of the 4 ES nodes which was a little bit different configured than the other 3 nodes. I added that node a year ago or so. 

Added now an email address which is only informed when such (seldom) errors occur to inform about the need of immediate action.
